modules/n1ql/pages/n1ql-language-reference/recursive.adoc

```diff
@@ -0,0 +1,315 @@
+= Recursive WITH Clause
+:description: Use the RECURSIVE keyword to enable recursive referencing in WITH clause common table expressions.
+:imagesdir: ../../assets/images
+:page-topic-type: reference
+
+[abstract]
+{description}
+
+[#purpose]
+== Purpose
+
+The common table expressions (CTEs) created by the WITH clause simplify complex queries and create temporary result sets which can be used as data sources or as expressions for queries.
+
+The SELECT definition for CTEs is split by the UNION/UNION ALL operators.
+
+* The left arm of UNION/UNION ALL contains an anchor clause, which is non-recursive. 
+The anchor clause produces an initial set of documents as an intermediate result set for the CTE keyspace.
+
+* The right arm of UNION/UNION ALL contains the recursive clause.
+The recursive clause produces a new set of documents.
+
+The recursive clause is executed again, with the CTE keyspace now referring to the result produced by the initial recursive clause.
+This process repeats until it results in an empty intermediate result set.
+The intermediate result sets are appended to produce a final output.
+
+[#syntax]
+== Syntax
+
+[source,ebnf]
+----
+include::partial$grammar/dql.ebnf[tag=recursive-with-clause]
+----
+
+image::n1ql-language-reference/recursive-with-clause.png["Syntax diagram", align=left]
+
+----
+include::partial$grammar/dql.ebnf[tag=recursive-with-clause-union]
```

Comment:
the new non-terminal -> ```recursive-with-clause-select```

Also we are **missing** the syntax diagrams for ```anchor-clause``` and ```recursive-clause```

anchor-clause would be a ```select```  from here: https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/select-syntax.html 

recursive-clause would be a ```select-term``` also from the above page.

Reply:
Do you mean we should add `recursive-with-clause-union ::= anchor_clause ( 'UNION' | 'UNION ALL' )? recursive_clause` to the "SELECT Syntax" page?

Reply:
Hi,
No, the sorry for the confusion with the syntax diagram
attaching an updated version -> https://drive.google.com/file/d/14_ZlLF2EDZxjZr9dyMWr2P3aAqFi3jX_/view?usp=sharing 

here the anchor clause and recursive clause rules
anchor_clause::= select  // the select can be hyperlinked to the select over here https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/select-syntax.html 

similarly , for recursive_clause ::= select-term // select-term can be hyperlinked to select-term here https://docs.couchbase.com/server/current/n1ql/n1ql-language-reference/select-syntax.html 

Also mention, select-term in the recursive clause will be marked as recursive only if it refers to the with-alias...

Reply:
This comment is mostly superseded by my suggested changes to the EBNF. We need to strike a balance between legibility and brevity 😉 

Incidentally, we _will_ have to update the SELECT syntax page, but let's leave that outside the scope of this PR for now.

Reply:
Sounds good! Thank you both!